internal/db/model.go

```diff
@@ -75,6 +77,49 @@ func (d *Database) GetModel(ctx context.Context, model *dbmodel.Model) (err erro
 	return nil
 }
 
+// SetModelMigrating updates the model's migration mode to the provided value.
+// This function will return an error if the model's migration mode is already
+// set to anything besides MigrationModeNone.
+func (d *Database) SetModelMigrating(ctx context.Context, uuid string, migrationMode dbmodel.MigrationMode) (m *dbmodel.Model, err error) {
```

Comment:
i'd suggest naming this `SetModelMigrationMode` or something similar as it takes an arbitrary migration mode